src/main/java/org/traccar/protocol/GatorProtocolDecoder.java

```diff
@@ -120,17 +176,99 @@ protected Object decode(
+                    }
+                }
+            }
+            if (type == MSG_ALARM_DATA) {
+                int alarmDataByte1 = buf.readUnsignedByte();
+                int alarmDataByte2 = buf.readUnsignedByte();
+
+                if ((alarmDataByte1 & 0b00100000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_GEOFENCE_EXIT);
+                }
+                if ((alarmDataByte1 & 0b00000001) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_GEOFENCE_ENTER);
+                }
+
+                if ((alarmDataByte2 & 0b10000000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_DOOR);
+                }
+                if ((alarmDataByte2 & 0b00100000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_VIBRATION);
+                }
+                if ((alarmDataByte2 & 0b00001000) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_POWER_OFF);
+                }
+                if ((alarmDataByte2 & 0b00000100) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_PARKING);
+                }
+                if ((alarmDataByte2 & 0b00000010) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_OVERSPEED);
+                }
+                if ((alarmDataByte2 & 0b00000001) >= 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_SOS);
+                }
+            }
+
+            return position;
+        } else if (type == MSG_TERMINAL_ANSWER) {
+            Position position = new Position(getProtocolName());
+
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, "1" + id, id);
+            if (deviceSession == null) {
+                return null;
+            }
+
+            position.setDeviceId(deviceSession.getDeviceId());
+            position.setTime(new Date());
+
+            getLastLocation(position, null);
+
+            int mainType = buf.readUnsignedByte();
+
+            // Sub Type - Skip 1 Byte
+            buf.readUnsignedByte();
+
+            int commandExecuted = buf.readUnsignedByte();
+
+            // Reserved - Skip 2 Bytes
+            buf.readUnsignedShort();
+
+            position.set(Position.KEY_EVENT, Event.TYPE_COMMAND_RESULT);
+
+            if (commandExecuted == 0x01) {
+                if (mainType == 0x38) {
+                    position.set(Position.KEY_COMMAND, Command.TYPE_ENGINE_RESUME);
+                    position.set(Position.KEY_RESULT, "Engine Started");
+                } else if (mainType == 0x39) {
+                    position.set(Position.KEY_COMMAND, Command.TYPE_ENGINE_STOP);
+                    position.set(Position.KEY_RESULT, "Engine Stopped");
+                }
+            } else {
+                position.set(Position.KEY_RESULT, "Command Failed");
+            }
```

Comment:
These all are also incorrect. In the `KEY_RESULT` we only include the original data from the device. You should not have hardcoded text there.

Reply:
Are you suggesting to remove **KEY_COMMAND** and **KEY_RESULT**?